app/controllers/prison/visits_controller.rb

```diff
@@ -18,14 +18,15 @@ def process_visit
   # rubocop:disable Metrics/MethodLength
   # rubocop:disable Metrics/AbcSize
   def update
-    @visit = load_visit.decorate
+    @visit = load_visit
     @visit.assign_attributes(visit_params)
     @booking_response = BookingResponse.new(visit: @visit, user: current_user)
     if @booking_response.valid?
       BookingResponder.new(@booking_response, message).respond!
       flash[:notice] = t('process_thank_you', scope: [:prison, :flash])
       redirect_to visit_page(@visit)
     else
+      @visit = @visit.decorate
```

Comment:
Will do.
Typically you only decorate your objects once right before render views. The README  https://github.com/drapergem/draper#when-to-decorate-objects as much better explanation
What happened here is that decorating the ```@visit``` object would also automatically decorate the related rejection object (as I have set it up). RejectionDecorator got passed down to ```BookingResponder::Reject``` processor and tried to save allowance_renews_on as an AccessibleDate with casted in an unexpected way.
I think that is what actually may have been wrong with test i marked as pending but i need to verify this.